src/main/resources/data/computercraft/lua/rom/apis/textutils.lua

```diff
@@ -335,6 +335,42 @@ empty_json_array = mk_tbl("[]", "empty_json_array")
 -- @see textutils.unserialiseJSON
 json_null = mk_tbl("null", "json_null")
 
+local serializeJSONString
+do
+    local function hexify(c)
+        return "\\u00" .. string.format("%02X", string.byte(c))
+    end
+
+    local hpat = string.format("[%c-%c]", 0x7F, 0xFF)
+
+    local map = {
+        ["\""] = "\\\"",
+        ["\\"] = "\\\\",
+        ["\b"] = "\\b",
+        ["\f"] = "\\f",
+        ["\n"] = "\\n",
+        ["\r"] = "\\r",
+        ["\t"] = "\\t",
+    }
+    for i = 0, 0x1f do
+        local c = string.char(i)
+        if map[c] == nil then map[c] = hexify(c) end
+    end
+
+    local function control_replace(c)
+        local r = map[c]
+        if r ~= nil then return r end
+        return c
+    end
+
+    serializeJSONString = function(s)
+        return string.format(
+            '"%s"',
+            s:gsub("[%c\"\\]", control_replace):gsub(hpat, hexify)
```

Comment:
I'm fairly sure we can do `:gsub("[%c\"\\]", map)` and drop the `control_replace` function.